src/lightning_app/runners/cloud.py

```diff
@@ -106,6 +106,90 @@ def _to_clean_dict(swagger_object, map_attributes):
 class CloudRuntime(Runtime):
     backend: Union[str, CloudBackend] = "cloud"
 
+    def open(self, name: str, cluster_id: Optional[str] = None):
+        """Method to open a CloudSpace with the root folder uploaded."""
+        try:
+            # Check for feature support
+            user = self.backend.client.auth_service_get_user()
+            if not user.features.code_tab:
+                print(
+                    "The `lightning open` command has not been enabled for your account. "
+                    "To request access, please contact support@lightning.ai"
+                )
+                sys.exit(1)
+
+            # Dispatch in four phases: resolution, validation, spec creation, API transactions
+            # Resolution
+            cloudspace_config = self._resolve_config(name, load=False)
+            root = self._resolve_root()
+            ignore_functions = self._resolve_open_ignore_functions()
+            repo = self._resolve_repo(root, ignore_functions)
+            project = self._resolve_project()
+            existing_cloudspaces = self._resolve_existing_cloudspaces(project, cloudspace_config.name)
+            cluster_id = self._resolve_cluster_id(cluster_id, project.project_id, existing_cloudspaces)
+            existing_cloudspace, existing_run_instance = self._resolve_existing_run_instance(
+                cluster_id, project.project_id, existing_cloudspaces
+            )
+            cloudspace_name = self._resolve_cloudspace_name(
+                cloudspace_config.name,
+                existing_cloudspace,
+                existing_cloudspaces,
+            )
+            needs_credits = self._resolve_needs_credits(project)
+
+            # Validation
+            # Note: We do not validate the repo here since open only uploads a directory if asked explicitly
+            self._validate_cluster_id(cluster_id, project.project_id)
+
+            # Spec creation
+            run_body = self._get_run_body(cluster_id, [], None, [], True, root, self.start_server)
+
+            if existing_run_instance is not None:
+                print(
+                    f"Re-opening the CloudSpace {cloudspace_config.name}. "
+                    "This operation will create a new run but will not update your files."
+                )
```

Comment:
what does "update files" mean? will it overwrite them only if they exist? remove them all and upload just these?

Reply:
Updated it to be more clear. Basically just warning that this won't actually update the files in your cloudspace, although we should probably add that